clients/src/main/java/org/apache/kafka/server/policy/TopicPolicy.java

```diff
@@ -0,0 +1,214 @@
+                     short currentReplicationFactor,
+                     Optional<Short> requestedReplicationFactor,
+                     Map<Integer, List<Integer>> replicaAssignments,
+                     Map<String, String> currentConfigs,
+                     Map<String, String> requestedConfigs) {
+            this.name = name;
+            this.currentPartitions = currentPartitions;
+            this.requestedPartitions = requestedPartitions;
+            this.currentReplicationFactor = currentReplicationFactor;
+            this.requestedReplicationFactor = requestedReplicationFactor;
+            this.replicaAssignments = replicaAssignments;
+            this.currentConfigs = currentConfigs;
+            this.requestedConfigs = requestedConfigs;
+        }
+
+        /** The name of the requested topic */
+        public String name() {
+            return name;
+        }
+
+        /**
+         * The current number of partitions.
+         * This will be 0 for a {@link Type#CREATE_TOPIC} request.
+         */
+        public int currentPartitions() {
+            return currentPartitions;
+        }
+
+        /**
+         * The requested number of partitions. The optional will be empty if a user defined replica assignment was supplied.
+         * This will be greater than {@link #currentPartitions()} for a {@link Type#CREATE_TOPIC} or
+         * {@link Type#CREATE_PARTITIONS} request.
+         */
+        public Optional<Integer> requestedPartitions() {
+            return requestedPartitions;
+        }
+
+        /**
+         * The current replication factor,
+         * defined as the number of replicas of the partition with the smallest partition index.
+         * This will be 0 for a {@link Type#CREATE_TOPIC} request.
+         */
+        public short currentReplicationFactor() {
+            return currentReplicationFactor;
+        }
+
+        /**
+         * The requested replication factor. This optional will be empty if a user defined replica assignment was supplied.
+         * This will be greater than {@link #currentPartitions()} for a {@link Type#CREATE_TOPIC} or
+         * {@link Type#REASSIGN_PARTITIONS} request.
+         */
+        public Optional<Short> requestedReplicationFactor() {
+            return requestedReplicationFactor;
+        }
+
+        /**
+         * The replica reassignments for this topic. This is in the form of a map from partition id to replica (broker) ids.
+         * This will be empty for anything other than {@link Type#CREATE_TOPIC} and {@link Type#REASSIGN_PARTITIONS} requests.
+         * For a {@link Type#CREATE_TOPIC} event this will be either populated with the user supplied replica assignment
+         * (in which case requestedPartitions and requestedReplicationFactor will be empty) the auto-generate assignment
+         * if partition count and/or replication factor were supplied.
+         */
+        public Map<Integer, List<Integer>> replicaAssignments() {
```

Comment:
Do we also want `requestedReplicaAssignments()`?

Reply:
Or do you mean we need to supply both the current replicaAssignment (empty for CreateTopic) and the requested?

Reply:
Yes the later. Otherwise what do you validate in a `REASSIGN_PARTITIONS` request?